public/app/percona/shared/services/advisors/Advisors.types.ts

```diff
@@ -0,0 +1,34 @@
+export enum Interval {
+  STANDARD = 'Standard',
+  RARE = 'Rare',
+  FREQUENT = 'Frequent',
+}
+
+export interface Advisor {
+  // Machine-readable name (ID) that is used in expression.
+  name: string;
+  // Long human-readable description.
+  description: string;
+  // Short human-readable summary.
+  summary: string;
+  // Comment.
+  comment: string;
+  // Category.
+  category: string;
+  // Advisor checks.
+  checks: AdvisorCheck[];
+}
+
+export interface AdvisorCheck {
+  name: string;
+  disabled: boolean;
+  description: string;
+  summary: string;
+  interval: keyof typeof Interval;
```

Comment:
Shouldn't the `Interval` enum be enough here instead of `keyof typeof`?

Reply:
Unfortunately it needs to be like this because of the way it is implemented in other parts of checks.